junit-jupiter-api/src/main/java/org/junit/jupiter/api/DisplayNameGenerator.java

```diff
@@ -412,26 +418,32 @@ private static DisplayNameGenerator getGeneratorFor(Class<?> testClass) {
 		/**
 		 * Find the first {@code DisplayNameGeneration} annotation that is either
 		 * <em>directly present</em>, <em>meta-present</em>, or <em>indirectly present</em>
-		 * on the supplied {@code testClass} or on an enclosing class.
+		 * on the supplied {@code testClass} or on an enclosing instance type.
 		 *
 		 * @param testClass the test class on which to find the annotation; never {@code null}
+		 * @param enclosingInstanceTypes the runtime types of the enclosing
+		 * instances; never {@code null}
 		 * @return an {@code Optional} containing the annotation, potentially empty if not found
 		 */
-		private static Optional<DisplayNameGeneration> findDisplayNameGeneration(Class<?> testClass) {
-			return findAnnotation(testClass, DisplayNameGeneration.class, SearchOption.INCLUDE_ENCLOSING_CLASSES);
+		@API(status = INTERNAL, since = "5.12")
+		static Optional<DisplayNameGeneration> findDisplayNameGeneration(Class<?> testClass,
```

Comment:
In the GitHub UI, I cannot determine if this method is used outside of this class.

In other words, does this need to be package private?

Reply:
Good catch! I had an interim state locally where I called it from `DisplayNameUtils` before introducing the new method in `AnnotationSupport`.

Made `private` again in 56ce4ffe3365c62e9c2680e80bee632879edba87